### PR TITLE
Feature/9/macro

### PR DESCRIPTION
**Pull request by judit-nahaj**

### CODE REVIEW

File: lib/builtIn/Macro.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+     * @returns {boolean|*} FALSE if the given element should be filtered out.
+     */
+    preFilterScenario(scenario, parent, i) {
+        const macroTag = scenario.tags.find(tag => /^@macro\(.*\)/.test(tag.name));
+
+        if (macroTag) {
+            const name = macroTag.name.substring(7, macroTag.name.length - 1);
+
+            if (name.length === 0) {
+                throw new Error('Name is not provided for macro for scenario ${scenario.name}.');
```
Comment (szikszail) [inline lib/builtIn/Macro.js]: @judit-nahaj I am sure, that here, you wanted to use the template literals, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: lib/builtIn/Macro.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            }
+
+            if (name in this.macros) {
+                throw new Error('Name ${name} already used in scenario ${this.macros[name].name}.');
```
Comment (szikszail) [inline lib/builtIn/Macro.js]: @judit-nahaj I am sure, that here, you wanted to use the template literals, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: lib/builtIn/Macro.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            }
+
+            if (scenario.steps.length === 0) {
+                throw new Error('Macro ${name} does not contain any steps.');
```
Comment (szikszail) [inline lib/builtIn/Macro.js]: @judit-nahaj I am sure, that here, you wanted to use the template literals, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: lib/builtIn/Macro.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            }
+
+            if (MACROSTEP.test(scenario.steps.map(step => step.text).join('\n'))) {
+                throw new Error('Macro ${name} contains a macro step.');
```
Comment (szikszail) [inline lib/builtIn/Macro.js]: @judit-nahaj I am sure, that here, you wanted to use the template literals, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: lib/builtIn/Macro.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+    onStep(step) {
+        if (MACROSTEP.test(step.text)) {
+            const name = step.text.match(MACROSTEP)[1].trim();
+            if (name.length === 0) {
+                throw new Error('Macro name is not provided for macro scenario.');
+            }
+
+            if (!(name in this.macros)) {
+                throw new Error('Macro ${name} does not exist.');
```
Comment (szikszail) [inline lib/builtIn/Macro.js]: @judit-nahaj I am sure, that here, you wanted to use the template literals, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

Comment (judit-nahaj) [reply lib/builtIn/Macro.js]: Fixed probably, would you check again please?

**Pull request merged**